java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -69,4 +83,8 @@ public Optional<String[]> getColumns() {
   public long getBatchSize() {
     return batchSize;
   }
+
+  public ByteBuffer getSubstraitExtendedExpression() {
```

Comment:
Why don't we separate out projection and filter? C++ does this.

Reply:
Yes, that will be an option. But currently we could define using the same extended expression:
- Or a projection
- Or a filter
- Or a projection + filter at the same time

I could divide that, then, 03 new options/methods will be created, or 01 methods with 03 new types.... Does it make sense?

Reply:
? You'd have `setProjection`, `setFilter`, and either or both could be null (and both are exclusive with `columns`)

Reply:
Thank you, added

Reply:
Please take another look at this. I'm not going to review this.

Reply:
There are: projection, filter and projectionAndFilter, that could by used at ScanOptions builder time, then this method is a helper needed for evaluate what of these option was used.

Could you help me to clarify what is your comment when you mention take another look on what is your recommendation please?

Reply:
Hey @davisusanibar , I think we should remove `getProjectionAndFilter` and `getSubstraitExtendedExpression`. If the user wants to set both, they can set filter and projection separately.